src/main/java/org/opentripplanner/routing/core/State.java

```diff
@@ -87,14 +93,22 @@ public State(Vertex vertex, long timeSeconds, RoutingRequest options) {
      * a RoutingContext in TransitIndex, tests, etc.
      */
     public State(Vertex vertex, Edge backEdge, long timeSeconds, RoutingRequest options) {
-        this(vertex, backEdge, timeSeconds, timeSeconds, options);
+        this(vertex, backEdge, timeSeconds, timeSeconds, options, false);
     }
-    
+
+    /**
+     * Create an initial state, forcing vertex, back edge and time to the specified values. Useful for reusing
+     * a RoutingContext in TransitIndex, tests, etc.
+     */
+    public State(Vertex vertex, Edge backEdge, long timeSeconds, RoutingRequest options, boolean carPickupStateInCar) {
+        this(vertex, backEdge, timeSeconds, timeSeconds, options, carPickupStateInCar);
+    }
+
     /**
      * Create an initial state, forcing vertex, back edge, time and start time to the specified values. Useful for starting
      * a multiple initial state search, for example when propagating profile results to the street network in RoundBasedProfileRouter.
      */
-    public State(Vertex vertex, Edge backEdge, long timeSeconds, long startTime, RoutingRequest options) {
+    public State(Vertex vertex, Edge backEdge, long timeSeconds, long startTime, RoutingRequest options, boolean carPickupStateInCar) {
```

Comment:
This parameter seems really specific for it to be included in the State constructor. I will comment more on it below.

Reply:
I agree that this could be organized better, but I didn't see a better option with the current code. The alternative would be to modify `getInitialStates()` to also set the value of `carPickup`, which doesn't seem any cleaner.